server/athenian/api/align/goals/dbaccess.py

```diff
@@ -71,3 +124,49 @@ async def _validate_goal_creation_info(
         raise GoalMutationError(
             f"Some teamId-s don't exist or access denied: {missing_repr}", HTTPStatus.FORBIDDEN,
         )
+
+
+async def _validate_team_goal_deletions(
+    account_id: int, goal_id: int, team_ids: Sequence[int], sdb_conn: DatabaseLike,
+) -> None:
+    where_clause = sa.and_(
+        Goal.account_id == account_id,
+        Team.owner_id == account_id,
+        TeamGoal.goal_id == goal_id,
+        TeamGoal.team_id.in_(team_ids),
+    )
+    select_stmt = sa.select(
+        [TeamGoal.team_id],
+    ).join_from(TeamGoal, Goal).join(Team).where(where_clause)
+    found = {row[0] for row in await sdb_conn.fetch_all(select_stmt)}
+    if missing := [team_id for team_id in team_ids if team_id not in found]:
+        missing_repr = ",".join(map(str, missing))
+        raise GoalMutationError(
+            f"TeamGoal-s to remove not found for teams: {missing_repr}", HTTPStatus.NOT_FOUND,
+        )
+
+
+async def _validate_team_goal_assignments(
+    account_id: int,
+    goal_id: int,
+    assignments: Sequence[TeamGoalTargetAssignment],
+    sdb_conn: Connection,
+) -> None:
+    goal_exists = await sdb_conn.fetch_val(sa.select([1]).where(sa.and_(
+        Goal.account_id == account_id, Goal.id == goal_id,
+    )))
+    if not goal_exists:
+        raise GoalMutationError(
+            f"Goal {goal_id} doesn't exist or access denied", HTTPStatus.NOT_FOUND,
+        )
+
+    teams_where_clause = sa.and_(
+        Team.id.in_(a.team_id for a in assignments), Team.owner_id == account_id,
+    )
+    teams_stmt = sa.select([Team.id]).where(teams_where_clause)
```

Comment:
```suggestion
    teams_stmt = sa.select([Team.id]).where(sa.and_(
        Team.id.in_(a.team_id for a in assignments),
        Team.owner_id == account_id,
    ))
```

In general, try to compress lines, it is easier to read and review more compact code. Here we avoid the extra mental pressure on the reviewer to remember the "teams_where_clause" variable.

Reply:
Done.
I would never intentionally put more mental pressure on the poor reviewer :)
By naming and outlining things, even if used once, I hope to help the reader by avoiding long statements and decomposing complexity in multiple statements.

Reply:
Haha, maybe I am too worn off by Sentry here. Variables are nice indeed, I fully agree. It's just that sometimes, this situation happens:
1. We crash on `sa.select`.
2. Error gets reported to Sentry.
3. Sentry takes the closest 10 variables to the line where the exception raised.
4. It orders it alphabetically and repr-s until it exceeds the size limit (4 or 8KB per stack frame, I think, it's not configurable).
5. There are garbage variables on top and they eat the whole quota so that, for example, `assignments` is not shown.
6. We cannot easily debug the error directly from Sentry.

![image](https://user-images.githubusercontent.com/2793551/169025001-405cf5f1-e584-41b1-a23c-b0c9944b6baa.png)